### ISSUE TITLE
Date Field permit value nil in params

### ISSUE THREAD

**Issue by patybastos** [TrestleAdmin/trestle#86]
Hi @spohlenz, this issue is a doubt.
I have a field with type `date_field` and he can have `null` value, i am using the trestle, however when editing an object that already had the date set and leave the field with null value, in the parameters comes the date that was already saved instead of null. Have any option that allows to return the parameter with the null value?

**Comment by svoboda-jan** [comment]
You can use the following workaround to add a link to clear the field:

```ruby
date_field :issued_at, append: '<a href="#" onclick="$(this).parents(\'.input-group\').find(\'input\').val(\'\'); return false;" style="color: red;"><i class="fa fa-close"></i></a>'.html_safe
```

**Comment by patybastos** [comment]
Thank you, i was adding a button to clean, but this suggestion is much better!!

**Comment by spohlenz** [comment]
I'll re-open this one as I feel there are two unsolved issues here:

1. **Bug fix**: A date picker field that is manually cleared out should not submit the original value with the form.
2. **Possible Enhancement:** Add some way to clear out the field without having to manually highlight and delete the date, perhaps as an 'X' within the field.

**Comment by spohlenz** [comment]
This is now handled automatically in c53956f.

<img width="352" alt="screen shot 2018-03-16 at 4 08 53 pm" src="https://user-images.githubusercontent.com/7600/37505383-8dc53562-2934-11e8-807e-88dc261aa51d.png">

My understanding of the bug in the previous comment was also incorrect. Flatpickr requires the enter key to be pressed before it will update the input value. I don't consider this ideal behavior but I will leave it as is for now.

**Comment by svoboda-jan** [comment]
@spohlenz:
- The presence of the clear link should not depend on `instance.config.allowInput`, when input is not allowed does not mean that clearing the field is not allowed. I think that a `data-X` attribute should be used for this.
- There is a `console.log` debug code left in the commit.

**Comment by svoboda-jan** [comment]
@spohlenz Also the defaults in `altFormat` are redundant since `Trestle.i18n` always provides a default value from the en locale. (discussed in https://github.com/TrestleAdmin/trestle/pull/89)

**Comment by spohlenz** [comment]
Thank you catching that @svoboda-jan. I have fixed this in 7827ddf and have added the `data-allow-clear` attribute in 6c07c65 to control the presence of the clear button.

I've decided I'll keep the i18n fallbacks, since it provides an additional fallback in case `config.javascript_i18n_keys` is altered incorrectly. I have also used the fallback pattern for other i18n values within the JS.

**Comment by svoboda-jan** [comment]
@spohlenz 👍
